v3iofs/fs.py

```diff
@@ -56,58 +54,70 @@ def __init__(self, v3io_api=None, v3io_access_key=None, **kw):
             transport_kind='requests',
         )
 
-    def ls(self, path, detail=True, **kwargs):
+    @classmethod
+    def _strip_protocol(cls, path):
+        """ Turn path from fully-qualified to file-system-specific
+        May require FS-specific handling, e.g., for relative paths or links.
+        """
+        path = stringify_path(path)
+        protos = (cls.protocol,) if isinstance(cls.protocol, str) else cls.protocol # NOQA
+        for protocol in protos:
+            path = path.rstrip("/")
+            if path.startswith(protocol + "://"):
+                path = path[len(protocol) + 3:]
+            elif path.startswith(protocol + ":"):
+                path = path[len(protocol) + 1:]
+        # use of root_marker to make minimum required path, e.g., "/"
+        if not path.startswith("/"):
+            path = f"/{path}"
+        return path or cls.root_marker
+
+    def _details(self, contents):
+        return [_details_of(c) for c in contents]
+
+    def ls(self, path, detail=False, **kwargs):
         """Lists files & directories under path"""
+        path = self._strip_protocol(path)
         full_path = path
         container, path = split_container(path)
         if not container:
             return self._list_containers(detail)
 
-        err = False
-        try:
```

Comment:
Why did you remove the `try/except` block?

Reply:
As written, the code demands the return of a HTTP.OK status.  I found it more straightforward to NOT raise an exception based on the HTTP response, but rather to inspect the response for the presence of both directories and files.